test/lint/test_runner/src/main.rs

```diff
@@ -31,7 +33,31 @@ fn get_git_root() -> String {
     check_output(git().args(["rev-parse", "--show-toplevel"])).unwrap()
 }
 
-fn lint_std_filesystem() -> Result<(), LintError> {
+fn lint_subtree() -> LintResult {
+    // This only checks that the trees are pure subtrees, it is not doing a full
+    // check with -r to not have to fetch all the remotes.
+    let mut good = true;
+    for subtree in [
+        "src/crypto/ctaes",
+        "src/secp256k1",
+        "src/minisketch",
+        "src/leveldb",
+        "src/crc32c",
+    ] {
+        good &= Command::new("test/lint/git-subtree-check.sh")
+            .arg(subtree)
+            .status()
+            .expect("command_error")
+            .success();
+    }
+    if good {
+        Ok(())
+    } else {
+        Err("".to_string())
```

Comment:
Maybe just add a "failure in subtree lint" (also for the others)?

Reply:
It's a bit jarring to see `Err("")`, but I guess there is little point in adding a parser for the script output, or duplicating the failure reporting, so this can just be improved if/when the lint checks are converted to Rust.